hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1190,6 +1246,37 @@ private void checkContainerOpen(KeyValueContainer kvContainer)
     throw new StorageContainerException(msg, result);
   }
 
+  /**
+   * Check if container is Closed. Throw exception otherwise.
+   * @param kvContainer
+   * @throws StorageContainerException
+   */
+  private void checkContainerClose(KeyValueContainer kvContainer)
+      throws StorageContainerException {
+
+    final State containerState = kvContainer.getContainerState();
+    if (containerState == State.QUASI_CLOSED || containerState == State.CLOSED || containerState == State.UNHEALTHY) {
+      return;
+    }
+
+    final ContainerProtos.Result result;
+    switch (containerState) {
+    case OPEN:
+    case CLOSING:
+    case RECOVERING:
+      result = UNCLOSED_CONTAINER_IO;
+      break;
+    case INVALID:
+      result = INVALID_CONTAINER_STATE;
+      break;
+    default:
+      result = CONTAINER_INTERNAL_ERROR;
+    }
+    String msg = "Requested operation not allowed as ContainerState is " +
+        containerState;
+    throw new StorageContainerException(msg, result);
```

Comment:
Can this method just return a `bool`? Throwing an exception to check the state of a container seems heavy handed. It makes sense for `writeChunkForClosedContainer` to throw an exception as the contract for that method is that it should be called only for closed containers. This method is just check on the state.

Reply:
This method is used in both `putBlockForClosedContainer` and `writeChunkForClosedContainer`, We shouldn't allow putBlock as well for non-closed container, hence the exception. We can change it to boolean and still fail the request, but it would be the same in the end.